client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client;
+
+import org.eclipse.hono.client.impl.CommandConsumer;
+import org.eclipse.hono.client.impl.ProtocolAdapterCommandConsumerFactoryImpl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+/**
+ * A factory for creating clients for the <em>AMQP 1.0 Messaging Network</em> to
+ * receive commands and send responses.
+ */
+public interface ProtocolAdapterCommandConsumerFactory extends ConnectionLifecycle<HonoConnection> {
+
+    /**
+     * Creates a new factory for an existing connection.
+     *
+     * @param connection The connection to the AMQP network.
+     * @param adapterInstanceId The id of the protocol adapter instance that this factory is running in.
+     * @return The factory.
+     * @throws NullPointerException if connection or gatewayMapper is {@code null}.
+     */
+    static ProtocolAdapterCommandConsumerFactory create(final HonoConnection connection, final String adapterInstanceId) {
+        return new ProtocolAdapterCommandConsumerFactoryImpl(connection, adapterInstanceId);
+    }
+
+    /**
+     * Initializes the ProtocolAdapterCommandConsumerFactory with the given components.
+     *
+     * @param commandTargetMapper The component for mapping an incoming command to the gateway (if applicable) and
+     *            protocol adapter instance that can handle it. Note that no initialization of this factory will be done
+     *            here, that is supposed to be done by the calling method.
+     * @param deviceConnectionClientFactory The factory to create a device connection client instance. Note that no
+     *            initialization of this factory will be done here, that is supposed to be done by the calling method.
+     */
+    void initialize(CommandTargetMapper commandTargetMapper,
```

Comment:
That's how it was before. Like I described in the ticket text above, taking care of the ConnectionLifecycle of the (contained) client factories inside the CommandConsumerFactory makes things complicated.
And taking care of the ConnectionLifecycle is needed as we have the client factory bean methods (in `AbstractAdapterConfig`) annotated with `@Scope("prototype")`, hence dealing with different client factory instances in the places they are used in the `AbstractAdapterConfig`. And as I understood, we want to keep the `@Scope("prototype")` annotation there for the case that the adapter is configured with `maxInstances > 1`, to have each `AbstractProtocolAdapterBase` verticle instance with its own set of client factory and more importantly connection object instances.

Reply:
Ok, so the biggest change seems to be that the CommandConsumerFactory now uses the protocol adapter's connections to the Device Registration and Device Connection services instead of using (and managing) its own ones, right?

Reply:
yes